### ISSUE TITLE
no-empty-file always false positives in @typescript-eslint/parser@6

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [sindresorhus/eslint-plugin-unicorn#2175]
Right now, `no-empty-file` assumes that if an AST node includes a `directive` property, then that's enough to indicate it's a directive:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/ab7732e1de0f431adc7316595dd5636781d33806/rules/no-empty-file.js#L9

That assumption no longer holds true in `@typescript-eslint/parser`'s v6.0.0 due to https://github.com/typescript-eslint/typescript-eslint/pull/6274. We changed the AST shape roughly like:

```diff
- directive?: string;
+ directive: string | undefined;
```

https://github.com/typescript-eslint/typescript-eslint/blob/df131e258c93e5714c88c0373cfeb2e1e75afc75/packages/ast-spec/src/statement/ExpressionStatement/spec.ts#L8

The PR explains why we did that (tl;dr: normalizing AST node shapes). So `eslint-plugin-unicorn` will need to update with a change like:

```diff
- const isDirective = node => node.type === 'ExpressionStatement' && 'directive' in node;
+ const isDirective = node => node.type === 'ExpressionStatement' && node.directive;
```

Right now, `no-empty-line` is flagging totally valid files such as https://github.com/jakebailey/pyright-action/pull/19/files#diff-1e058ca1442e46581b13571fb8d261f9e1f5657e26c96634d4c1072f0f0347f1. Thanks @jakebailey for reporting!

**Comment by muuvmuuv** [comment]
Seeing this in all my test files where just a describe is in

**Comment by muuvmuuv** [comment]
Tested @JoshuaKGoldberg suggested changes and we no longer get these error, here's a patch for `eslint-plugin-unicorn@48.0.0` (used pnpm):

```diff
diff --git a/rules/no-empty-file.js b/rules/no-empty-file.js
index 858e0d008f773d5237f0fb919e82d048aefffbdb..9fa6699d74f62eba66e98c9fb0809164d046f97e 100644
--- a/rules/no-empty-file.js
+++ b/rules/no-empty-file.js
@@ -6,7 +6,7 @@ const messages = {
 	[MESSAGE_ID]: 'Empty files are not allowed.',
 };
 
-const isDirective = node => node.type === 'ExpressionStatement' && 'directive' in node;
+const isDirective = node => node.type === 'ExpressionStatement' && node.directive;
 const isEmpty = node => isEmptyNode(node, isDirective);
 
 const isTripleSlashDirective = node =>
```

**Comment by yakov116** [comment]
Why don't you submit a PR here?

**Comment by yoyo837** [comment]
Same here.

**Comment by mcous** [comment]
I opened #2180 to resolve this issue! If you're working around this issue with the patch above in the meantime, please be aware that it is not quite right, as the `directive` field may be defined but falsy, e.g. an empty string
